data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/node/GetPeerCount.java

```diff
@@ -63,7 +102,24 @@ public GetPeerCount(final DataProvider provider, final JsonProvider jsonProvider
   @Override
   public void handle(@NotNull final Context ctx) throws Exception {
     ctx.header(Header.CACHE_CONTROL, CACHE_NONE);
-    List<Peer> peers = network.getPeers();
-    ctx.json(jsonProvider.objectToJSON(GetPeerCountResponse.create(peers)));
+    adapt(ctx);
+  }
+
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
+    List<Eth2Peer> peers = network.getEth2Peers();
+    request.respondOk(new ResponseData(peers));
+  }
+
+  static class ResponseData {
+    private final List<Eth2Peer> data;
+
+    ResponseData(List<Eth2Peer> peers) {
+      this.data = peers;
+    }
+
+    List<Eth2Peer> getData() {
+      return data;
+    }
```

Comment:
we should maybe make use of this response object to reduce double handling...

on creation we could count the connected / not connected, store that value, and just query the value in the type...